src/windows/models/emoji_model.py

```diff
@@ -153,68 +158,45 @@ def update_model(self, clear=True):
 
                 row = []
 
-                # Append thumbnail
-                col = QStandardItem()
-                col.setIcon(QIcon(thumb_path))
-                col.setText("%s (%s)" % (emoji_name, emoji_type))
-                col.setToolTip(emoji_name)
-                col.setData(emoji_type)
-                col.setFlags(Qt.ItemIsSelectable | Qt.ItemIsEnabled | Qt.ItemIsUserCheckable | Qt.ItemIsDragEnabled)
-                row.append(col)
-
-                # Append Filename
+                # Set emoji data
                 col = QStandardItem("Name")
-                col.setData(emoji_name, Qt.DisplayRole)
+                col.setIcon(QIcon(thumb_path))
                 col.setText(emoji_name)
+                col.setToolTip(emoji_name)
+                col.setData(path)
                 col.setFlags(Qt.ItemIsSelectable | Qt.ItemIsEnabled | Qt.ItemIsUserCheckable | Qt.ItemIsDragEnabled)
                 row.append(col)
```

Comment:
Same thing here. This loads up the first column of the model with everything needed to define an item — the Name, the icon, and the filesystem path (in the UserRole as set by `setData()`).

By doing that, when we get QModelIndex entries for a selected or dragged item, we can query all that info directly from the individual item — there's no need to translate back to rows in the source model so we can match up siblings. All of the data is right where we need it.